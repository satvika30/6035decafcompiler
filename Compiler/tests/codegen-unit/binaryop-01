class Program
{
    void main()
    {
   		int a, b, c;
   		callout("printf", "a: %d\n", a);
   		callout("printf", "b: %d\n", b);
   		callout("printf", "c: %d\n", c);

        a = 10;
        b = 20;

        c = a + b;

        callout("printf", "c = a + b, c: %d\n", c);
        
        c = a - b;
        
        callout("printf", "c = a - b, c: %d\n", c);
        
        c = a - 10;

        callout("printf", "c = a - 10, c: %d\n", c);
    }
}
