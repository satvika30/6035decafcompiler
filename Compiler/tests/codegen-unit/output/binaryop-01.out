a: 0
b: 0
c: 0
c = a + b, c: 30
c = a - b, c: -10
c = a - 10, c: 0
